pkg/kubelet/dockertools/docker.go

```diff
@@ -117,29 +117,22 @@ type dockerContainerCommandRunner struct {
 }
 
 // The first version of docker that supports exec natively is 1.3.0 == API 1.15
-var dockerAPIVersionWithExec = []uint{1, 15}
+var dockerAPIVersionWithExec, _ = docker.NewAPIVersion("1.15")
 
 // Returns the major and minor version numbers of docker server.
-func (d *dockerContainerCommandRunner) GetDockerServerVersion() ([]uint, error) {
+func (d *dockerContainerCommandRunner) GetDockerServerVersion() (docker.APIVersion, error) {
 	env, err := d.client.Version()
 	if err != nil {
 		return nil, fmt.Errorf("failed to get docker server version - %v", err)
 	}
-	version := []uint{}
-	for _, entry := range *env {
-		if strings.Contains(strings.ToLower(entry), "apiversion") || strings.Contains(strings.ToLower(entry), "api version") {
-			elems := strings.Split(strings.Split(entry, "=")[1], ".")
-			for _, elem := range elems {
-				val, err := strconv.ParseUint(elem, 10, 32)
-				if err != nil {
-					return nil, fmt.Errorf("failed to parse docker server version %q: %v", entry, err)
-				}
-				version = append(version, uint(val))
-			}
-			return version, nil
-		}
+
+	apiVersion := env.Get("ApiVersion")
```

Comment:
I checked back to Docker 1.0 and it seemed consistent. All but 1.6 below were built using `AUTO_GOPATH=1 ./hack/make.sh dynbinary`.

1.0.0: `{"ApiVersion":"1.12","Arch":"amd64","GitCommit":"63fe64c","GoVersion":"go1.4.2","KernelVersion":"3.18.9-200.fc21.x86_64","Os":"linux","Version":"1.0.0"}`

1.4.0:
`{"ApiVersion":"1.16","Arch":"amd64","GitCommit":"4595d4f","GoVersion":"go1.4.2","KernelVersion":"3.18.9-200.fc21.x86_64","Os":"linux","Version":"1.4.0"}`

1.5.0:
`{"ApiVersion":"1.17","Arch":"amd64","GitCommit":"a8a31ef","GoVersion":"go1.4.2","KernelVersion":"3.18.9-200.fc21.x86_64","Os":"linux","Version":"1.5.0"}`

1.6.0:
`{"ApiVersion":"1.18","Arch":"amd64","GitCommit":"0591dce/1.6.0","GoVersion":"go1.4.2","KernelVersion":"3.18.9-200.fc21.x86_64","Os":"linux","Version":"1.6.0"}`